File: DSA/Manav-Rajput/Day23_question2.java
public class Solution {
    public double myPow(double x, int n) {
        if (n == 0) {
            return 1.0;
        }
        double result = 1.0;
        long absN = Math.abs((long)n);  
        while (absN > 0) {
            if ((absN & 1) == 1) {
                result *= x;
            }
            x *= x;
            absN >>= 1;
        }

        return n < 0 ? 1.0 / result : result;
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.myPow(2.0, 10));   
        System.out.println(solution.myPow(2.1, 3));    
        System.out.println(solution.myPow(2.0, -2));   
    }
}

File: DSA/Pragya_Agarwal/Day23/Day23_Question2.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0:
           return 1
        if n <0:
           x = 1/x
           n= -n
        result=1 
        prod = x

        while n > 0:
            if n % 2 == 1: 
               result *= prod
            prod *= prod
            n //= 2  
        return result 
